src/solver/opamActionGraph.ml

```diff
@@ -111,22 +124,34 @@ module MakeAction (P: GenericPackage) : ACTION with type package = P.t
         (P.version_to_string p)
 
   let to_aligned_strings ?(append=(fun _ -> "")) l =
+    let pkg_to_string ?(append=append) p =
+      [P.name_to_string p; P.version_to_string p ^ append p ]
+    in
     List.map (fun a ->
         let a = (a :> package action) in
         (if OpamConsole.utf8 ()
          then action_color a (symbol_of_action a)
          else "-")
         :: name_of_action a
-        :: OpamConsole.colorise `bold
-          (P.name_to_string (OpamTypesBase.action_contents a))
         :: match a with
-        | `Remove p | `Install p | `Reinstall p | `Build p | `Fetch p ->
-          (P.version_to_string p ^ append p) :: []
+        | `Remove p | `Install p | `Reinstall p | `Build p | `Fetch [p] ->
+          pkg_to_string p
+        | `Fetch (p::pl) ->
+          let append _ = "," in
+          let rec aux str p = function
+            | [] -> pkg_to_string p::str
+            | p'::pl -> aux (pkg_to_string ~append p::str) p' pl
+          in
+          aux [] p pl
+          |> List.rev
+          |> List.flatten
```

Comment:
I don't think it matters since this function is used for printing the actions summary, which doesn't include fetches; nevertheless, this should remain a single "table cell" (with a `String.concat " "`) rather than multiple table cells. Especially since we will be appending more columns to the right (for action reasons), the returned `list list` is expected to be a matrix, i.e. all rows of the same length

Reply:
noted!